### PR TITLE
Break Update Function Code into its own dedicated UI

### PR DESCRIPTION
**Pull request by abrooksv**
* Make Update Function Code have its own UI instead of bunch of hidden elements since it can lead to hidden errors
* Break up our panels
* Move validation logic closer to the UI

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/UpdateFunctionCodeDialog.kt [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                is Exception -> {
+                    error.notifyError(project = project, title = NOTIFICATION_TITLE)
+                    LambdaTelemetry.editFunction(project, update = false, result = Result.Failed)
+                }
+            }
+        }
+    }
+
+    // Using an OkAction to force the validation logic to trigger as well
+    private inner class UpdateFunctionOkAction : OkAction() {
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/UpdateFunctionCodeDialog.kt:120]: nit: curious why named inner class vs anonymous object?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/EditFunctionDialog.kt [hunk 2]
```diff
@@ -56,7 +56,7 @@ import javax.swing.JComponent
 private val NOTIFICATION_TITLE = message("lambda.service_name")
 
 enum class EditFunctionMode {
-    NEW, UPDATE_CONFIGURATION, UPDATE_CODE
+    NEW, UPDATE_CONFIGURATION, @Deprecated("All code paths using this should be dead") UPDATE_CODE
```
[the remark this answers is not in the document]
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/EditFunctionDialog.kt:59]: Probably 2 more long term

Update Function Configuration
Create new

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/UpdateFunctionCodeDialog.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/UpdateFunctionCodeDialog.kt:120]: It was inner class before as well, now that there isnt a switch statement powering it, probably fine to go anonymous.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CodeStoragePanel.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        content.border = IdeBorderFactory.createTitledBorder(message("lambda.upload.deployment_settings"), false)
+        add(content, BorderLayout.CENTER)
+    }
+
+    private fun createUIComponents() {
+        sourceBucket = ResourceSelector.builder().resource(listBucketNamesByActiveRegion(project)).awsConnection(project).build()
+    }
+
+    fun validatePanel(): ValidationInfo? {
+        if (sourceBucket.selected() == null) {
```
[the remark this answers is not in the document]
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CodeStoragePanel.kt:48]: Added

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CodeStoragePanel.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                    sourceBucket.selectedItem = it
+                }
+            }
+        }
+
+        content.border = IdeBorderFactory.createTitledBorder(message("lambda.upload.deployment_settings"), false)
+        add(content, BorderLayout.CENTER)
+    }
+
+    private fun createUIComponents() {
```
[the remark this answers is not in the document]
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CodeStoragePanel.kt:43]: I think I am in the consistent layout of classes camp

**Comment by hunterwerlla** [comment]
The test failure seems real as well:
```

s.services.lambda.upload.UpdateFunctionCodeDialogTest | Loads saved settings if function arn matches | FAILED
-- | -- | --
org.opentest4j.AssertionFailedError:  Expecting:  <"Loading..."> to be equal to:  <"hello2">


```

**Comment by abrooksv** [comment]
> The test failure seems real as well:
> 
> ```
> 
> s.services.lambda.upload.UpdateFunctionCodeDialogTest | Loads saved settings if function arn matches | FAILED
> -- | -- | --
> org.opentest4j.AssertionFailedError:  Expecting:  <"Loading..."> to be equal to:  <"hello2">
> ```

yep, seems the new still loading message broke the test

**Comment by abrooksv** [comment]
> The test failure seems real as well:
> 
> ```
> 
> s.services.lambda.upload.UpdateFunctionCodeDialogTest | Loads saved settings if function arn matches | FAILED
> -- | -- | --
> org.opentest4j.AssertionFailedError:  Expecting:  <"Loading..."> to be equal to:  <"hello2">
> ```

Had to add retry assert...

ResourceSelector switches to LOADED state before we set the value. This is because setting the value will trigger any actionListener. We can't switch it around because otherwise anything that tries to react to the change in the actionListener will always get null since we would still be in LOADING state...

**Review by hunterwerlla — APPROVED** [review]

**Pull request merged**